include/RAJA/policy/sycl/launch.hpp

```diff
@@ -0,0 +1,1170 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   RAJA header file containing user interface for RAJA::launch::sycl
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-22, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_pattern_launch_sycl_HPP
+#define RAJA_pattern_launch_sycl_HPP
+
+#include "RAJA/pattern/launch/launch_core.hpp"
+#include "RAJA/pattern/detail/privatizer.hpp"
+#include "RAJA/policy/sycl/policy.hpp"
+#include "RAJA/policy/sycl/MemUtils_SYCL.hpp"
+//#include "RAJA/policy/sycl/raja_syclerrchk.hpp"
+#include "RAJA/util/resource.hpp"
+
+namespace RAJA
+{
+
+template <bool async>
+struct LaunchExecute<RAJA::sycl_launch_t<async, 0>> {
+
+  //If the launch lambda is trivially copyable
+  template <typename BODY_IN,
+	    typename std::enable_if<std::is_trivially_copyable<BODY_IN>{},bool>::type = true>
+  static void exec(size_t shared_mem_size, LaunchContext const &ctx, BODY_IN &&body_in)
+  {
+
+    cl::sycl::queue* q = ::RAJA::sycl::detail::getQueue();
+
+    resources::Sycl sycl_res = resources::Sycl::get_default();
+
+    // Global resource was not set, use the resource that was passed to forall
+    // Determine if the default SYCL res is being used
+    if (!q) {
+      q = sycl_res.get_queue();
+    }
+
+    const ::sycl::range<3> blockSize(ctx.threads.value[0],
+				     ctx.threads.value[1],
+				     ctx.threads.value[2]);
+
+    const ::sycl::range<3> gridSize(ctx.threads.value[0] * ctx.teams.value[0],
+				    ctx.threads.value[1] * ctx.teams.value[1],
+				    ctx.threads.value[2] * ctx.teams.value[2]);
+
+        // Only launch kernel if we have something to iterate over
+    constexpr size_t zero = 0;
+    if ( ctx.threads.value[0]  > zero && ctx.threads.value[1]  > zero && ctx.threads.value[2] > zero &&
+         ctx.teams.value[0] > zero && ctx.teams.value[1] > zero && ctx.teams.value[2]> zero ) {
+
+      RAJA_FT_BEGIN
+
+      q->submit([&](cl::sycl::handler& h) {
+
+        auto s_vec = cl::sycl::accessor<char, 1, cl::sycl::access::mode::read_write,
+                                        cl::sycl::access::target::local> (shared_mem_size, h);
+
+        h.parallel_for
+          (cl::sycl::nd_range<3>(gridSize, blockSize),
+           [=] (cl::sycl::nd_item<3> itm) {
+
+             ctx.itm = &itm;
+
+             //Point to shared memory
+             ctx.shared_mem_ptr = s_vec.get_pointer().get();
```

Comment:
Does it make more sense to make this lambda mutable so you don't have to make the members of LaunchContext mutable?